openmc/tallies.py

```diff
@@ -259,24 +264,69 @@ def with_summary(self):
 
     @property
     def sum(self):
+        if not self._statepoint:
+            return None
+
+        if not self._results_read:
+            # Extract Tally data from the file
+            data = self._statepoint._f['tallies/tally {0}/results'.format(
+                self.id)].value
+            sum = data['sum']
+            sum_sq = data['sum_sq']
+
+            # Define a routine to convert 0 to 1
+            def nonzero(val):
+                return 1 if not val else val
+
+            # Reshape the results arrays
+            new_shape = (nonzero(self.num_filter_bins),
+                         nonzero(self.num_nuclides),
+                         nonzero(self.num_score_bins))
+
+            sum = np.reshape(sum, new_shape)
+            sum_sq = np.reshape(sum_sq, new_shape)
+
+            # Set the data for this Tally
+            self._sum = sum
+            self._sum_sq = sum_sq
+
+            # Indicate that Tally results have been read
+            self._results_read = True
+
         return self._sum
 
     @property
     def sum_sq(self):
+        if not self._statepoint:
+            return None
+
+        if not self._results_read:
+            # Force reading of sum and sum_sq
+            self.sum
+
         return self._sum_sq
 
     @property
     def mean(self):
-        # Compute the mean if needed
         if self._mean is None:
-            self.compute_mean()
+            if not self._statepoint:
+                return None
+
+            self._mean = self.sum / self.num_realizations
         return self._mean
 
     @property
     def std_dev(self):
-        # Compute the standard deviation if needed
         if self._std_dev is None:
-            self.compute_std_dev()
+            if not self._statepoint:
+                return None
+
+            n = self.num_realizations
+            nonzero = np.abs(self.mean) > 0
+            self._std_dev = np.zeros_like(self.mean)
+            self._std_dev[nonzero] = np.sqrt((self.sum_sq[nonzero]/n -
+                                              self.mean[nonzero]**2)/(n - 1))
+            self.with_batch_statistics = True
```

Comment:
If you decide to load tally data on the fly and return it here, the ``with_batch_statistics`` property will no longer be relevant.